internal/provider/resource_virtual_machine_test.go

```diff
@@ -583,6 +592,105 @@ func TestAccKatapultVirtualMachine_update_ips(t *testing.T) {
 	})
 }
 
+func TestAccKatapultVirtualMachine_update_group(t *testing.T) {
+	tt := newTestTools(t)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy: resource.ComposeAggregateTestCheckFunc(
+			testAccCheckKatapultVirtualMachineDestroy(tt),
+			testAccCheckKatapultIPDestroy(tt),
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: undent.String(`
+					resource "katapult_ip" "web" {}
+
+					resource "katapult_virtual_machine_group" "web" {
+						name = "vm-group"
+					}
+
+					resource "katapult_virtual_machine" "base" {
+						package       = "rock-3"
+						disk_template = "ubuntu-18-04"
+						disk_template_options = {
+							install_agent = true
+						}
+						ip_address_ids = [
+							katapult_ip.web.id,
+						]
+					}`,
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKatapultVirtualMachineExists(
+						tt, "katapult_virtual_machine.base",
+					),
+					resource.TestCheckNoResourceAttr(
+						"katapult_virtual_machine.base",
+						"group_id",
+					),
+				),
+			},
+			{
+				Config: undent.String(`
+					resource "katapult_ip" "web" {}
+
+					resource "katapult_virtual_machine_group" "web" {
+						name = "vm-group"
+					}
+
+					resource "katapult_virtual_machine" "base" {
+						package       = "rock-3"
+						disk_template = "ubuntu-18-04"
+						disk_template_options = {
+							install_agent = true
+						}
+						ip_address_ids = [
+							katapult_ip.web.id,
+						]
+						group_id = katapult_virtual_machine_group.web.id
+					}`,
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKatapultVirtualMachineExists(
+						tt, "katapult_virtual_machine.base",
+					),
+					resource.TestCheckResourceAttrPair(
+						"katapult_virtual_machine.base", "group_id",
+						"katapult_virtual_machine_group.web", "id",
+					),
+				),
+			},
+			{
+				Config: undent.String(`
+				resource "katapult_ip" "web" {}
+
+				resource "katapult_virtual_machine" "base" {
+					package       = "rock-3"
+					disk_template = "ubuntu-18-04"
+					disk_template_options = {
+						install_agent = true
+					}
+					ip_address_ids = [
+						katapult_ip.web.id,
+					]
+				}`,
```

Comment:
Indentation is off by one here :)

```suggestion
					resource "katapult_ip" "web" {}

					resource "katapult_virtual_machine" "base" {
						package       = "rock-3"
						disk_template = "ubuntu-18-04"
						disk_template_options = {
							install_agent = true
						}
						ip_address_ids = [
							katapult_ip.web.id,
						]
					}`,
```